projects/netgrif-components-core/src/lib/authorization/role/role-guard.service.ts

```diff
@@ -39,31 +39,28 @@ export class RoleGuardService implements CanActivate {
     public canAccessView(view: View, url: string): boolean {
         if (typeof view.access !== 'string' && (view.access.hasOwnProperty('role') || view.access.hasOwnProperty('bannedRole'))) {
 
-            if (view.access.hasOwnProperty('bannedRole')) {
+            if (view.access.hasOwnProperty('role') && view.access.hasOwnProperty('bannedRole')) {
                 const bannedRoles = this.parseRoleConstraints(view.access.bannedRole, url);
+                const allowedRoles = this.parseRoleConstraints(view.access.role, url);
+                return allowedRoles.some(constraint => {return this.decideAccessByRole(constraint)}) &&
+                    ((bannedRoles.length == 0) ||
+                        (bannedRoles.length > 0 &&
+                            !bannedRoles.some(constraint => {return this.decideAccessByRole(constraint)})));
+            }
 
+            if (view.access.hasOwnProperty('bannedRole')) {
+                const bannedRoles = this.parseRoleConstraints(view.access.bannedRole, url);
                 return (bannedRoles.length == 0) || (bannedRoles.length > 0 && !bannedRoles.some(constraint => {
```

Comment:
`return !bannedRoles.some(role => this.decideAccessByRole(role));` is sufficient, because `some` returns `false` on an empty array and the negation turns it to `true` which is the desired behavior.

Reply:
Done